doc/porting_guide.md

```diff
@@ -3,268 +3,284 @@
+	make clean
+	make clean_all
 
-`cd examples`
+##### Building the Examples
+From the base directory of the repository:
 
-`make`
+	cd examples
+	make
 
-here if previously library was built with mbedTLS setting the XI_BSP_TLS variable is necessary to tell the linker which TLS lib to link agains:
+Here if the library was previously built with the mbedTLS setting then the XI_BSP_TLS variable is necessary.  Otherwise you will get link errors.
 
-`make XI_BSP_TLS=mbedTLS`
+	make XI_BSP_TLS=mbedTLS
 
 
-#### Build dependencies
+#### Build Dependencies
 
 - autotools: autoconf, automake, libtool are needed by wolfSSL
 - cmake is needed by the integration test framework _cmocka_
 
-On OSX these are available through _brew_:
+On OSX these are available through `brew`:
 
     - brew install autoconf
     - brew install automake
     - brew install libtool
-    - brew install cmake
+    - brew install cmak
+
+On Ubuntu these are available through `apt-get`:
+	
+	- sudo apt-get update
+	- sudo apt-get install autoconf
+	- sudo apt-get install autotools-dev
+	- sudo apt-get install libtool
+   
 
 #### Xively C Client Build Flags
 
-The make system uses dash '-' separated string to define the TARGET and CONFIGuration
-for a build. TARGET defines the build environment for specific platforms and toolchains.
-The collection of CONFIG flags specify selections from a list of Xively Client modules and features to be included in the Xively C Client library.
+The make system uses dash '-' separated string to define the TARGET and CONFIGuration for a build. 
+
+TARGET defines the build environment for specific platforms and toolchains.
+
+CONFIG flags specify selections from a list of Xively Client modules and features to be included in the Xively C Client library.
 
 _Default_ values will be chosen for you if you do not set the TARGET or CONFIG makefile variables.
 
-The TARGET and CONFIG flags are enumerated below, and can be mixed and matched in any order and are separated by the - (minus sign / dash) character.
+The TARGET and CONFIG flags are enumerated below, and can be mixed and matched in any order.  Each flag isseparated by the - (minus sign / dash) character.
 
-##### TARGET flags
+##### TARGET Flags
 
 A typical TARGET flag looks like this:
 
     osx-static-debug
 
-###### Platform flag
+###### Platform Flag
 
     - [ osx | linux | arm-linux ]   - selects the target platform
 
-###### Output Library type
+###### Output Library Type
 
     - [ static | dynamic ]          - selects the output library type
 
 ###### Build Type
 
     - [ debug | release ]           - selects the build type
 
-##### CONFIG flags
+##### CONFIG Flags
 
 A typical CONFIG flag:
 
-    posix_io-posix_fs-thread_module-posix_platform-tls-senml-control_topic-memory_limiter
+    posix_io-posix_fs-thread_module-posix_platform-tls-senml-memory_limiter
 
 ###### Xively Client Feature flags
 
     - control_topic     - the control topic feature is in development.
```

Comment:
actually it's used by the file system updates.
so yes, I think within a year it will be used!